x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/CompositeAggRowSet.java

```diff
@@ -49,7 +49,7 @@ static int remainingData(boolean hasNextPage, int size, int limit) {
         if (hasNextPage == false) {
             return 0;
         } else {
-            int remainingLimit = (limit == -1) ? limit : ((limit - size) >= 0 ? (limit - size) : 0);
+            int remainingLimit = limit == -1 ? limit : Math.max(limit - size, 0);
```

Comment:
Change done to compact the expression and match the similar construct in the constructor (above).